backends/guest/common/validate.c

```diff
@@ -260,7 +260,7 @@ int validate_esl(const uint8_t *esl_data, size_t esl_data_len)
 	if (!count)
 		return ESL_FAIL;
 
-	return SUCCESS;
+	return rc;
```

Comment:
Agreed. Changed to reducing the scope of `rc` instead of persisting it through the whole function. There's no test case that failed for this, but there really should be if this change of behavior is supposed to be important.

Reply:
This is a weird function though -- why do we handled what seems to be the first one different from the second. Why is a function called `validate_esl` validating a list, and should it not be breaking and erroring if we're validating a list of esls and one fails? I don't know what the right answer is here, but this is slated for rework.